jwst/cube_build/file_table.py

```diff
@@ -1,155 +1,158 @@
+        self.FileMap["MIRI"]["4"]["medium"] = []
+        self.FileMap["MIRI"]["4"]["long"] = []
+        self.FileMap["MIRI"]["4"]["short-medium"] = []
+        self.FileMap["MIRI"]["4"]["short-long"] = []
+        self.FileMap["MIRI"]["4"]["medium-short"] = []
+        self.FileMap["MIRI"]["4"]["medium-long"] = []
+        self.FileMap["MIRI"]["4"]["long-short"] = []
+        self.FileMap["MIRI"]["4"]["long-medium"] = []
+
+        self.FileMap["NIRSPEC"] = {}
+        self.FileMap["NIRSPEC"]["prism"] = {}
+        self.FileMap["NIRSPEC"]["prism"]["clear"] = []
+        self.FileMap["NIRSPEC"]["prism"]["opaque"] = []
+
+        self.FileMap["NIRSPEC"]["g140m"] = {}
+        self.FileMap["NIRSPEC"]["g140m"]["f070lp"] = []
+        self.FileMap["NIRSPEC"]["g140m"]["f100lp"] = []
+        self.FileMap["NIRSPEC"]["g140m"]["opaque"] = []
+
+        self.FileMap["NIRSPEC"]["g140h"] = {}
+        self.FileMap["NIRSPEC"]["g140h"]["f070lp"] = []
+        self.FileMap["NIRSPEC"]["g140h"]["f100lp"] = []
+        self.FileMap["NIRSPEC"]["g140h"]["opaque"] = []
+
+        self.FileMap["NIRSPEC"]["g235m"] = {}
+        self.FileMap["NIRSPEC"]["g235m"]["f170lp"] = []
+        self.FileMap["NIRSPEC"]["g235m"]["opaque"] = []
+
+        self.FileMap["NIRSPEC"]["g235h"] = {}
+        self.FileMap["NIRSPEC"]["g235h"]["f170lp"] = []
+        self.FileMap["NIRSPEC"]["g235h"]["opaque"] = []
+
+        self.FileMap["NIRSPEC"]["g395m"] = {}
+        self.FileMap["NIRSPEC"]["g395m"]["f290lp"] = []
+        self.FileMap["NIRSPEC"]["g395m"]["opaque"] = []
+
+        self.FileMap["NIRSPEC"]["g395h"] = {}
+        self.FileMap["NIRSPEC"]["g395h"]["f290lp"] = []
+        self.FileMap["NIRSPEC"]["g395h"]["opaque"] = []
+
+    # ********************************************************************************
+
+    def set_file_table(self, input_models):
         """
-        Short Summary
-        -------------
+        Set up the MasterTable dictionary.
+
         Fill in the MasterTable which holds the files that the cube will be constructed
-        from. Since MIRI has 2 channels per image this MASTERTable helps to figure out
+        from. Since MIRI has 2 channels per image this MASTERTable dictionary helps to figure out
         which data needs to be use.
-        THe MasterTable for MIRI is broken down by channel and subchannel.
+        The MasterTable for MIRI is broken down by channel and subchannel.
         For each channel/subchannel combination - a file is listed that covers those options
-        For NIRSPEC the table contains the Grating and Filter for each file.
+        For NIRSPEC the table contains the grating and filter for each file.
 
+        Parameters
+        ----------
+        input_models : IFUImageModel
+           The input datamodels used the set up the class FileTable
 
         Returns
         -------
-        MasterTable filled in with files needed
-        instrument name
+        MasterTable : dict
```

Comment:
The output name should also be lower case:
```suggestion
        master_table : dict
```